### PR TITLE
Upload example of transforming raw MERFISH -> spaceTx formatted data

### PR DESCRIPTION
**Pull request by dganguli**
Remaining work includes:
1. Running this tool against all raw data
2. Uploading formatted data to cloud
3. Using experiment API against this dataset to test parsing larger datasets

### CODE REVIEW

**Comment by dganguli** [comment]
@joshmoore -- take a look at the raw tiff files on s3. I suspect there's metadata in them that can be used to obviate my explicit map. also curious what you think! will work on an nd2 example shortly.

**Comment by joshmoore** [comment]
* Ran `get_merfish*py`
 * Renamed `for x in hybridization-fov*tiff; do mv $x ${x//H/T}; done` as a workaround
 * Created a file with the contents `hybridization-fov_000-Z0-T<0-7>-C<0-1>.tiff` file & loaded in Bio-Formats

screen shot 2018-08-24 at 12 42 13

i.e. if the input files had been named as above with the one extra file, the FOV writer would have generated the same output (assuming my code is correct, which I'll now verify...)

You stopped at 16 (154 MB) of the 496 (71 GB) TIFFs to just get this working ASAP? (Or is something else going on? S3 didn't contain an XLS for cross-checking.

**Comment by joshmoore** [comment]
With the [last commit](https://gitlab.com/openmicroscopy/incubator/spacetx-fov-writer/commit/79cbfe749beba4a4684f3e2465bf7b483a9e2088) on spacetx-fov-writer and ignoring file names and checksums, here's the diff:

```
$ diff -w <(grep -vE "(file|sha256)" /tmp/merfish-deep/hybridization-fov_000.json) <(grep -vE "(file|sha256)" /tmp/merfish-josh/hybridization-fov_000.json)
14c14,16
<     "extras": {},
---
>     "extras": {
>       "OME": "hybridization-fov_000.companion.ome"
>     },
438c440
<     "version": "0.0.0"
---
>     "version": "1.0.0"
```

**Comment by ambrosejcarr** [comment]
@joshmoore You work fast. 😮

File: examples/get_merfish_u20s_data.py [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+from starfish.types import Indices
+from starfish.util.argparse import FsExistsType
+
+SHAPE = 2048, 2048
+
+
+class MERFISHTile(FetchedTile):
+    def __init__(self, file_path, hyb, ch):
+        self.file_path = file_path
+        # how to index tiles from indices into multi-page tiff
```
Comment (ambrosejcarr) [inline examples/get_merfish_u20s_data.py]: I want to make sure I'm clear on what this means: 
`# maps information on the acquisition channel and round to an index that identifies the correct page in the original multi-page tiff file`

**Comment by joshmoore** [comment]
> You work fast.

You guys have to sleep at _some_ point!

**Comment by ambrosejcarr** [comment]
> You guys have to sleep at some point!

Says you.

File: examples/get_merfish_u20s_data.py (hunk 1 above)
Comment (ambrosejcarr) [reply examples/get_merfish_u20s_data.py]: Thanks.

**Comment by dganguli** [comment]
_I did some digging, @dganguli, but didn't find any metadata. Do you know anything about the provenance of these files?_

Really? Nothing fun embedded in the tiffs? I do not know anything about the provenance of these files.

**Comment by dganguli** [comment]
_You stopped at 16 (154 MB) of the 496 (71 GB) TIFFs to just get this working ASAP? (Or is something else going on? S3 didn't contain an XLS for cross-checking.)_

Oops the 16 is a bug. Fixing. I'll also add the XLS to s3.

**Comment by dganguli** [comment]
@joshmoore : s3://czi.starfish.data.public/browse/raw/20180820/merfish_u2os/data_organization.xlsx

is the xls file to check against.

**Comment by joshmoore** [comment]
> is the xls file to check against.

Cheers.

**Comment by joshmoore** [comment]
> Really? Nothing fun embedded in the tiffs? I do not know anything about the provenance of these files.

```
$ tiffinfo ...
TIFF Directory at offset 0x9000d14 (150998292)
  Image Width: 2048 Image Length: 2048
  Bits/Sample: 16
  Compression Scheme: None
  Photometric Interpretation: min-is-black
  Planar Configuration: single image plane
  Software: MATLAB
```

My guess is going to be that the generating MATLAB script didn't store anything.

For those following along, here's the contents of the spreadsheet:

```
In [8]: print(df.to_string())
         0                             1                                                  2          3             4      5      6                            7                                                  8                     9              10             11
0   bitName                     imageType                                        imageRegExp  bitNumber  imagingRound  color  frame            fiducialImageType                                     fiducialRegExp  fiducialImagingRound  fiducialFrame  fiducialColor
1    RS0015      Conventional_750_650_560  (?<imageType>\w+)_(?<fov>[0-9]+)_(?<imagingRou...          1             1    650      2     Conventional_750_650_560  (?<imageType>\w+)_(?<fov>[0-9]+)_(?<imagingRou...                     1              3            561
2    RS0083      Conventional_750_650_560  (?<imageType>\w+)_(?<fov>[0-9]+)_(?<imagingRou...          2             1    750      1     Conventional_750_650_560  (?<imageType>\w+)_(?<fov>[0-9]+)_(?<imagingRou...                     1              3            561
3    RS0095      Conventional_750_650_560  (?<imageType>\w+)_(?<fov>[0-9]+)_(?<imagingRou...          3             2    750      1     Conventional_750_650_560  (?<imageType>\w+)_(?<fov>[0-9]+)_(?<imagingRou...                     2              3            561
4    RS0109      Conventional_750_650_560  (?<imageType>\w+)_(?<fov>[0-9]+)_(?<imagingRou...          4             2    650      2     Conventional_750_650_560  (?<imageType>\w+)_(?<fov>[0-9]+)_(?<imagingRou...                     2              3            561
5    RS0175      Conventional_750_650_560  (?<imageType>\w+)_(?<fov>[0-9]+)_(?<imagingRou...          5             3    650      2     Conventional_750_650_560  (?<imageType>\w+)_(?<fov>[0-9]+)_(?<imagingRou...                     3              3            561
6    RS0237      Conventional_750_650_560  (?<imageType>\w+)_(?<fov>[0-9]+)_(?<imagingRou...          6             3    750      1     Conventional_750_650_560  (?<imageType>\w+)_(?<fov>[0-9]+)_(?<imagingRou...                     3              3            561
7    RS0247      Conventional_750_650_560  (?<imageType>\w+)_(?<fov>[0-9]+)_(?<imagingRou...          7             4    650      2     Conventional_750_650_560  (?<imageType>\w+)_(?<fov>[0-9]+)_(?<imagingRou...                     4              3            561
8    RS0255      Conventional_750_650_560  (?<imageType>\w+)_(?<fov>[0-9]+)_(?<imagingRou...          8             4    750      1     Conventional_750_650_560  (?<imageType>\w+)_(?<fov>[0-9]+)_(?<imagingRou...                     4              3            561
9    RS0307      Conventional_750_650_560  (?<imageType>\w+)_(?<fov>[0-9]+)_(?<imagingRou...          9             5    750      1     Conventional_750_650_560  (?<imageType>\w+)_(?<fov>[0-9]+)_(?<imagingRou...                     5              3            561
10   RS0332      Conventional_750_650_560  (?<imageType>\w+)_(?<fov>[0-9]+)_(?<imagingRou...         10             5    650      2     Conventional_750_650_560  (?<imageType>\w+)_(?<fov>[0-9]+)_(?<imagingRou...                     5              3            561
11   RS0343      Conventional_750_650_560  (?<imageType>\w+)_(?<fov>[0-9]+)_(?<imagingRou...         11             6    650      2     Conventional_750_650_560  (?<imageType>\w+)_(?<fov>[0-9]+)_(?<imagingRou...                     6              3            561
12   RS0384      Conventional_750_650_560  (?<imageType>\w+)_(?<fov>[0-9]+)_(?<imagingRou...         12             6    750      1     Conventional_750_650_560  (?<imageType>\w+)_(?<fov>[0-9]+)_(?<imagingRou...                     6              3            561
13   RS0406      Conventional_750_650_560  (?<imageType>\w+)_(?<fov>[0-9]+)_(?<imagingRou...         13             7    650      2     Conventional_750_650_560  (?<imageType>\w+)_(?<fov>[0-9]+)_(?<imagingRou...                     7              3            561
14   RS0451      Conventional_750_650_560  (?<imageType>\w+)_(?<fov>[0-9]+)_(?<imagingRou...         14             7    750      1     Conventional_750_650_560  (?<imageType>\w+)_(?<fov>[0-9]+)_(?<imagingRou...                     7              3            561
15   RS0468      Conventional_750_650_560  (?<imageType>\w+)_(?<fov>[0-9]+)_(?<imagingRou...         15             8    750      1     Conventional_750_650_560  (?<imageType>\w+)_(?<fov>[0-9]+)_(?<imagingRou...                     8              3            561
16   RS0548      Conventional_750_650_560  (?<imageType>\w+)_(?<fov>[0-9]+)_(?<imagingRou...         16             8    650      2     Conventional_750_650_560  (?<imageType>\w+)_(?<fov>[0-9]+)_(?<imagingRou...                     8              3            561
17      WGA  CostainImages_1_488_405_fast                   (?<imageType>\w+)_(?<fov>[0-9]+)         17            -1    488      1  CostainImages_2_750_650_560                   (?<imageType>\w+)_(?<fov>[0-9]+)                    -1              3            561
18     DAPI  CostainImages_1_488_405_fast                   (?<imageType>\w+)_(?<fov>[0-9]+)         18            -1    405      2  CostainImages_2_750_650_560                   (?<imageType>\w+)_(?<fov>[0-9]+)                    -1              3            561
```

**Comment by joshmoore** [comment]
With:
 * https://github.com/spacetx/sptx-format/pull/19
 * https://github.com/spacetx/sptx-format/pull/20
 * Manual creation of a (fake) codebook
 * Manual addition of "fov" values:

```
$ grep '"fov": 0' /data/merfish-2018082-a/*.json
/data/merfish-2018082-a/hybridization-fov_000.json:    "fov": 0,
/data/merfish-2018082-a/nuclei-fov_000.json:    "fov": 0
```

I was able to validate this dataset:

```
$ docker run --rm -v /data/merfish-2018082-a/:/data sptx-format --experiment-json /data/experiment.json && echo yay
yay
```

**Comment by joshmoore** [comment]
@dganguli @ttung : `fov`s provisionally added in https://github.com/spacetx/slicedimage/pull/55

**Pull request merged**